### PR TITLE
Add tests to CI pipeline

### PR DESCRIPTION
**Pull request by dchiquit**
Add a new job to the CircleCI pipeline that will install the girder app
and the web app, run them, then run the tests against them.

For now, I'm pointing this PR at `291-puppeteer-tests`. As soon as that is merged I will re-aim this at `refactor` (or `master`, if we've done the final merge at that point).

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -36,6 +36,53 @@ jobs:
...
+          command: |
+            python3 -m venv install_env
+            . install_env/bin/activate
+      - run:
+          name: Upgrade pip
+          command: pip install -U pip
+      - run:
+          name: Install girder-dandi-archive
+          command: |
+            whereis python
```
Comment (brianhelba) [inline .circleci/config.yml:57]: Remove.

File: .circleci/config.yml [hunk 2]
```diff
@@ -36,6 +36,53 @@ jobs:
           command: yarn run build
           working_directory: web
 
+  test-end-to-end:
+    docker:
+      - image: circleci/python:3.6-node-browsers
+      - image: circleci/mongo:4.2
```
Comment (brianhelba) [inline .circleci/config.yml:42]: Note, if we use this a lot, we can refactor it out as a CircleCI executor.

File: .circleci/config.yml [hunk 3]
```diff
@@ -36,6 +36,53 @@ jobs:
           command: yarn run build
```
Comment (brianhelba) [inline .circleci/config.yml:36]: Cut the production build here, since we're doing a dev build in e2e.

**Comment by yarikoptic** [comment]
I can't believe you like tests too. We should hang out. [*](https://www.imdb.com/title/tt0387808/)

**Comment by brianhelba** [comment]
@dchiquit I think you need to rebase this, to get CircleCI to pick up the updated non-`refactor` config.

**Comment by brianhelba** [comment]
I just merged `master` into this.

**Pull request merged**